tools/report-converter/codechecker_report_converter/sparse/output_parser.py

```diff
@@ -0,0 +1,106 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import re
+
+from ..output_parser import BaseParser, Message, Event
+LOG = logging.getLogger('ReportConverter')
+
+
+class SparseParser(BaseParser):
+    """
+    Parser for Sparse Output
+    """
+
+    def __init__(self, analyzer_result):
+        super(SparseParser, self).__init__()
+
+        self.analyzer_result = analyzer_result
+
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*')
+
+        self.multi_line_message_re = re.compile(
+            r'^(?P<path>[\S ]+?):'
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+        self.note_line_re = re.compile(
+            r'^(?P<path>\.[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+    def parse_message(self, it, line):
+        """
+        Actual Parsing function for the given line
+        It is expected that each line contains a seperate report
+        """
+        match = self.message_line_re.match(line)
+        multi_line_match = self.multi_line_message_re.match(line)
+
+        if (match is None) and (multi_line_match is None):
+            return None, next(it)
+
+        checker_name = None
+
+        if match:
+            file_path = os.path.normpath(
+                os.path.join(os.path.dirname(self.analyzer_result),
+                             match.group('path')))
+            message = Message(
+                file_path,
+                int(match.group('line')),
+                int(match.group('column')),
+                match.group('message').strip(),
+                checker_name)
+
+        else:
+            file_path = os.path.normpath(
+                os.path.join(os.path.dirname(self.analyzer_result),
+                             multi_line_match.group('path')))
+
+            message = Message(
+                file_path,
+                0,
```

Comment:
The GUI will not able to show this message because the line is zero and not 1:
![image](https://user-images.githubusercontent.com/6695818/108344888-f0682400-71dd-11eb-9205-0a6e71ce7177.png)
This is the use case when the source line looks like this: `files/sample.c: note: in included file:`. I think in this case we can skip this line because it doesn't contain too much information for the developer.

My recommendation is to do the following:
- Every line which looks like this will be a new report (Message): `./files/sample.h:3:40: warning: incorrect type in argument 2 (different address spaces)` (the `warning` keyword identifies that we have a new report).
- Every line which looks like this will be an event / bug step item of the previous Message object: `./files/sample.h:3:40:    expected struct task_struct *p1`. (So it doesn't contain the warning keyword like in the previous case).
- Every line can be skipped which looks like this `files/sample.c: note: in included file:`.

What do you think?

Reply:
Well if we are skipping every line which looks like : `files/sample.c: note: in included file` then we can simply remove:
the regex for multi line re. Cause the main report lines start with `files/sample.c` and event lines start with `./files/sample.c`